### PR TITLE
Preference Subscreen Escaping

### PR DESCRIPTION
**Pull request by Nina-1474**
When in a preferences subscreen, hitting the Esc key exited the preferences menus entirely whereas it's much more intuitive to jump back one screen to the preferences home page.

Also much of the subscreen exit logic was contained in Click functions, which meant that exiting via the Esc key found no Exit function and therefore didn't perform the proper cleanup. These steps have now all been extracted into `PreferenceSubscreen*Exit()` functions so they're always run.

Also also I refactored all MouseX and MouseY `if` statements in the file into MouseIns, and cleaned up indents/whitespaces/semicolons.

### CODE REVIEW

File: BondageClub/Screens/Character/Preference/Preference.js [hunk 1]
```diff
@@ -1315,11 +1327,8 @@ function PreferenceSubscreenAudioExit() {
  * @returns {void} - Nothing
  */
 function PreferenceSubscreenControllerClick() {
-    if ((MouseX >= 1815) && (MouseX < 1905) && (MouseY >= 75) && (MouseY < 165)) {
-        PreferenceSubscreen = "";
-        PreferenceCalibrationStage = 0;
-        Calibrating = false;
-    }
+	if ((MouseX >= 1815) && (MouseX < 1905) && (MouseY >= 75) && (MouseY < 165)) PreferenceSubscreenControllerExit();
```
Comment (jomshir98) [inline BondageClub/Screens/Character/Preference/Preference.js:1330]: `MouseIn`?

File: BondageClub/Screens/Character/Preference/Preference.js [hunk 2]
```diff
@@ -1399,8 +1408,7 @@ function PreferenceSubscreenChatClick() {
 	}
 
 	// If the user clicked the exit icon to return to the main screen
-	if ((MouseX >= 1815) && (MouseX < 1905) && (MouseY >= 75) && (MouseY < 165) && (PreferenceColorPick == ""))
-		PreferenceSubscreen = "";
+	if ((MouseX >= 1815) && (MouseX < 1905) && (MouseY >= 75) && (MouseY < 165)) PreferenceSubscreenChatExit();
```
Comment (jomshir98) [inline BondageClub/Screens/Character/Preference/Preference.js:1411]: `MouseIn` here too?

File: BondageClub/Screens/Character/Preference/Preference.js [hunk 3]
```diff
@@ -1433,10 +1441,7 @@ function PreferenceSubscreenOnlineClick() {
 function PreferenceSubscreenArousalClick() {
 
 	// If the user clicked the exit icon to return to the main screen
-	if ((MouseX >= 1815) && (MouseX < 1905) && (MouseY >= 75) && (MouseY < 165)) {
-		PreferenceSubscreen = "";
-		Player.FocusGroup = null;
-	}
+	if ((MouseX >= 1815) && (MouseX < 1905) && (MouseY >= 75) && (MouseY < 165)) PreferenceSubscreenArousalExit();
```
Comment (jomshir98) [inline BondageClub/Screens/Character/Preference/Preference.js:1444]: `MouseIn` again...

File: BondageClub/Screens/Character/Preference/Preference.js [hunk 4]
```diff
@@ -1538,11 +1543,7 @@ function PreferenceSubscreenArousalClick() {
 function PreferenceSubscreenSecurityClick() {
 
 	// If the user clicked the exit icon to return to the main screen
-	if ((MouseX >= 1815) && (MouseX < 1905) && (MouseY >= 75) && (MouseY < 165)) {
-		PreferenceSubscreen = "";
-		ElementRemove("InputEmailOld");
-		ElementRemove("InputEmailNew");
-	}
+	if ((MouseX >= 1815) && (MouseX < 1905) && (MouseY >= 75) && (MouseY < 165)) PreferenceSubscreenSecurityExit();
```
Comment (jomshir98) [inline BondageClub/Screens/Character/Preference/Preference.js:1546]: Few more of `MouseIn`s

**Review by jomshir98 — COMMENTED** [review]
And there are some more places you didn't change, that could use `MouseIn` too~

**Comment by Nina-1474** [comment]
@jomshir98 Yeah the entire file is littered with MouseX/Y checks that should be MouseIns and I was just focusing on the exit functions. 
.... Soooo now I've changed the lot~

**Comment by Ben987** [comment]
Sure!  Thanks for that one.  Cheers :)

**Pull request merged**